pkg/sentry/kernel/thread_group.go

```diff
@@ -516,6 +511,11 @@ func (tg *ThreadGroup) SetForegroundProcessGroup(tty *TTY, pgid ProcessGroupID)
 		return -1, linuxerr.EPERM
 	}
 
+	//if the calling process is a member of a background group, a SIGTTOU signal is sent to all members of this background process group.
+	if tg.processGroup.session.foreground.id != tg.processGroup.id {
```

Comment:
I believe we also need to check whether the calling process (`tg` here) is blocking or ignoring `SIGTTOU`:

> If tcsetpgrp() is called by a member of a background process group in its session, **and the calling process is not blocking or ignoring SIGTTOU**, a SIGTTOU signal is sent to all members of this background process group.

This is also something worth testing.

Reply:
> I believe we also need to check whether the calling process (`tg` here) is blocking or ignoring `SIGTTOU`:
> 
> > If tcsetpgrp() is called by a member of a background process group in its session, **and the calling process is not blocking or ignoring SIGTTOU**, a SIGTTOU signal is sent to all members of this background process group.
> 
> This is also something worth testing.

I'm sorry, I assumed that if the process neglects the signal, the latter isn't sent at all. I did add a check to verify if the process is ignoring the SIGTTOU and another test as well for this case (SetForegroundProcessGroupSIGTTOUIgnored).

Reply:
There should also be a check for whether the process is blocking SIGTTOU, which can be done via [`task.SignalMask()`](https://cs.opensource.google/gvisor/gvisor/+/master:pkg/sentry/kernel/task_signals.go;l=587;bpv=0;bpt=0). I believe you can reuse the test you wrote for ignoring to also check for blocking.